core/src/main/scala/pureconfig/CoproductHint.scala

```diff
@@ -92,6 +92,37 @@ class FieldCoproductHint[T](key: String) extends CoproductHint[T] {
 }
 
 /**
+ * Hint applicable to sealed families of case objects where objects are written and read as strings. Trying to read or
+ * write values that are not case objects results in failure.
+ *
+ * @tparam T the type of the coproduct or sealed family for which this hint applies
+ */
+class EnumCoproductHint[T] extends CoproductHint[T] {
+
+  /**
+   * Returns the field value for a class or coproduct option name.
+   *
+   * @param name the name of the class or coproduct option
+   * @return the field value associated with the given class or coproduct option name.
+   */
+  protected def fieldValue(name: String): String = name.toLowerCase
```

Comment:
Yes, it follows the same structure as `FieldCoproductHint`, where the translation between class names and config values is customizable.

Just as we did with `ProductHint` (following [the discussion here](https://github.com/pureconfig/pureconfig/pull/90#discussion_r97449022)), we could provide a constructor for both of these hints that receives a `ConfigFieldMapping`. I'd rather keep a consistent interface between hints here and leave a possible refactoring or new constructor for another PR.